s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/FileReader.java

```diff
@@ -62,11 +62,19 @@ Iterator<S3ObjectSummary> fetchObjectSummaries(final AmazonS3 s3Client) {
                 .flatMap(response -> response.getObjectSummaries()
                         .stream()
                         .filter(objectSummary -> objectSummary.getSize() > 0)
+                        .filter(objectSummary -> assignObjectToTask(objectSummary.getKey()))
                         .filter(objectSummary -> !failedObjectKeys.contains(objectSummary.getKey())));
         return s3ObjectStream.iterator();
     }
 
     public void addFailedObjectKeys(final String objectKey) {
         this.failedObjectKeys.add(objectKey);
     }
+
+    private boolean assignObjectToTask(final String objectKey) {
+        final int maxTasks = Integer.parseInt(s3SourceConfig.originals().get("tasks.max").toString());
+        final int taskId = Integer.parseInt(s3SourceConfig.originals().get("task.id").toString()) % maxTasks;
+        final int taskAssignment = Math.floorMod(objectKey.hashCode(), maxTasks);
+        return taskAssignment == taskId;
```

Comment:
As I was looking into this for task Assignment for distributed mode I noticed that there is a bug here.

floorMod will give you a mod for 4 tasks between 0-3 but tasks are identified as 1-4 so we will need to account for that here.

There are also no unit tests related to this change yet

Reply:
Sorry i've been hold up fighting gradle all day.

Ok I see that taskid is being mod'd as well which also brings it to 0 here.
So taskAssignment and the task id both end up at 0.

Reply:
yea, just pushed another fix, where task id starts with zero too.